test/suite/hook_tiered.py

```diff
@@ -0,0 +1,145 @@
+# to create. The default is 2.
+#
+# To run with 3 tiers per table:
+#     ../test/suite/run.py --hooks tiered=3 cursor
+#
+# The hooks work with may other tests in the python suite but also encounter
+# a variety of failures that I haven't tried to sort out.
+from __future__ import print_function
+
+import os, sys, wthooks
+import unittest
+from wttest import WiredTigerTestCase
+
+# Print to /dev/tty for debugging, since anything extraneous to stdout/stderr will
+# cause a test error.
+def tty(s):
+    WiredTigerTestCase.tty(s)
+
+# These are the hook functions that are run when particular APIs are called.
+
+# Called to replace Session.create
+def session_create_replace(arg, orig_session_create, session_self, uri, config):
+    if config == None:
+        base_config = ""
+    else:
+        base_config = config
+
+    # If the test is creating a table (not colstore or lsm), create a tiered table instead,
+    # using arg to determine number of tiers. Otherwise just do the create as normal.
+    #
+    # NOTE: the API for creating a tiered table will change in WT-7173 and this will need to
+    # be updated.
+    if uri[:6] == "table:" and "key_format=r" not in base_config and "type=lsm" not in base_config:
+        tier_string = ""
+        for i in range(arg):
+            new_uri = uri.replace('table:', 'file:tier' + str(i) + '_')
+            ret = orig_session_create(session_self, new_uri, config)
+            if ret != 0:
+                return ret
+            tier_string = tier_string + '"' + new_uri + '", '
+        tier_config = 'type=tiered,tiered=(tiers=(' + tier_string[0:-2] + ')),' + base_config
+        WiredTigerTestCase.verbose(None, 3,
+            'Creating tiered table {} with config = \'{}\''.format(uri, tier_config))
+        ret = orig_session_create(session_self, uri, tier_config)
+    else:
+        ret = orig_session_create(session_self, uri, config)
+    return ret
+
+# Called to replace Session.drop
+def session_drop_replace(arg, orig_session_drop, session_self, uri, config):
+    # Drop isn't implemented for tiered tables.  Only do the delete if this could be a
+    # uri we created a tiered table for.  Note this isn't a precise when we did/didn't
+    # create a tiered table, but we don't have the create config around to check.
+    ret = 0
+    if (uri[:6] != 'table:'):
+        ret = orig_session_drop(session_self, uri, config)
+    return ret
+
+# Called to replace Session.verify
+def session_verify_replace(arg, orig_session_verify, session_self, uri):
+    return 0
+
+# Every hook file must have one or more classes descended from WiredTigerHook
+# This is where the hook functions are 'hooked' to API methods.
+class TieredHookCreator(wthooks.WiredTigerHookCreator):
+    def __init__(self, arg=0):
+        # Argument specifies the number of tiers to test. The default is 2.
+        if arg == None:
+            self.arg = 2
+        else:
+            self.arg = int(arg)
```

Comment:
Probably good to use a more descriptive name for `self.arg`, maybe `self.ntiers`?
And wherever `args` is passed as a parameter to hook functions, use `ntiers` or whatever.